metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_api.py

```diff
@@ -0,0 +1,213 @@
+from ssl import CERT_NONE, PROTOCOL_TLSv1_2, SSLContext
+from typing import List, Optional
+
+from cassandra import DriverException, OperationTimedOut
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import (
+    EXEC_PROFILE_DEFAULT,
+    Cluster,
+    ExecutionProfile,
+    ProtocolVersion,
+    Session,
+)
+
+from datahub.ingestion.api.source import SourceReport
+from datahub.ingestion.source.cassandra.cassandra_config import CassandraSourceConfig
+from datahub.ingestion.source.cassandra.cassandra_utils import (
+    CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES,
+    CassandraQueries,
+)
+
+
+class CassandraAPIInterface:
+    def __init__(self, config: CassandraSourceConfig, report: SourceReport):
+        self.config = config
+        self.report = report
+        self.cassandra_session = self.authenticate()
+
+    def authenticate(self) -> Session:
+        """Establish a connection to Cassandra and return the session."""
+        try:
+            if self.config.cloud:
+                cloud_config = self.config.datastax_astra_cloud_config
+                assert cloud_config
+                cluster_cloud_config = {
+                    "connect_timeout": cloud_config.connect_timeout,
+                    "use_default_tempdir": True,
+                    "secure_connect_bundle": cloud_config.secure_connect_bundle,
+                }
+                profile = ExecutionProfile(request_timeout=cloud_config.request_timeout)
+                auth_provider = PlainTextAuthProvider(
+                    "token",
+                    cloud_config.token,
+                )
+                cluster = Cluster(
+                    cloud=cluster_cloud_config,
+                    auth_provider=auth_provider,
+                    execution_profiles={EXEC_PROFILE_DEFAULT: profile},
+                    protocol_version=ProtocolVersion.V4,
+                )
+
+                session: Session = cluster.connect()
+                return session
+
+            auth_provider = None
+            ssl_context = None
+            if self.config.username and self.config.password:
+                ssl_context = SSLContext(PROTOCOL_TLSv1_2)
+                ssl_context.verify_mode = CERT_NONE
+                auth_provider = PlainTextAuthProvider(
+                    username=self.config.username, password=self.config.password
+                )
+
+            cluster = Cluster(
+                [self.config.contact_point],
+                port=self.config.port,
+                load_balancing_policy=None,
+            )
+
+            if auth_provider:
+                cluster = Cluster(
+                    [self.config.contact_point],
+                    port=self.config.port,
+                    auth_provider=auth_provider,
+                    ssl_context=ssl_context,
+                    load_balancing_policy=None,
+                )
```

Comment:
Can we please move this to `if self.config.username and self.config.password:` block above and keep the init without auth_provider in else block ? this will help improve readability.

Reply:
Yes